internal/metrics/metrics.go

```diff
@@ -249,35 +250,44 @@ var (
 		[]string{"chain_name", "upstream_id", "url", "method"},
 	)
 
-	errorLatencyStatusCheckRequests = promauto.NewCounterVec(
+	errorLatencyStatusCheckErrors = promauto.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: metricsNamespace,
 			Subsystem: "healthcheck",
-			Name:      "latency_check_requests",
-			Help:      "Total latency check requests made.",
+			Name:      "error_check_has_errors",
+			Help:      "Number of errors of upstream requests.",
 		},
-		[]string{"chain_name", "upstream_id", "url", "method"},
+		[]string{"chain_name", "upstream_id", "url", "errorType", "method"},
 	)
 
-	errorLatencyStatusCheckDuration = promauto.NewHistogramVec(
-		prometheus.HistogramOpts{
+	errorLatencyStatusCheckNoErrors = promauto.NewCounterVec(
```

Comment:
nit: these variable names don't need to be `errorLatency` anymore, right? can they be either `error` or `latency`?